src/App.tsx

```diff
@@ -1,15 +1,47 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [chosenPerson, setChosenPerson] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
+
+  const handleQueryChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(e.target.value);
+    applyQuery(e.target.value);
+    setChosenPerson('');
+  };
```

Comment:
The naming convention for event handlers in React suggests that handlers should be named starting with 'handle' and ending with the name of the event. In your case, 'handleQueryChange' is a good example of following this convention. However, 'onSelected' does not follow this convention. Consider renaming it to 'handleSelection' or something similar. You can refer to [this article](https://medium.com/javascript-in-plain-english/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364) for more information on naming conventions in React.